language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -455,25 +455,23 @@ void setClientCapabilities(LSClientCapabilities clientCapabilities) {
     @Override
     public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
         return CompletableFuture.supplyAsync(() -> {
-            WorkspaceEdit workspaceEdit = new WorkspaceEdit();
-
             try {
                 RenameContext context = ContextBuilder.buildRenameContext(params.getTextDocument().getUri(),
                         this.workspaceManager,
                         this.serverContext,
                         params.getPosition());
+                context.setHonorsChangeAnnotations(this.clientCapabilities.getTextDocCapabilities()
+                        .getRename().getHonorsChangeAnnotations());
```

Comment:
Shall we move this to the constructor? Let's pass the rename parameters to the context builder and also let's save these RenameParams within the RenameContext. Explicitly setting can lead to errors in the long run if we are going to adhere changes.